### DISCUSSION TITLE
Computing grid velocities by finite differencing-SU2 has diverged (NaN detected)

### DISCUSSION THREAD

**Discussion by cucss77** [su2code/SU2#2408] [Q&A]
I have to simulate an inflation of a parachute. I provided a surface_positions_00001.dat file but I get this error. Why?

% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
%
SOLVER= EULER
KIND_TURB_MODEL= NONE
MATH_PROBLEM= DIRECT
RESTART_SOL= YES
RESTART_ITER=1
READ_BINARY_RESTART= NO
% ------------------------- Time-dependent Simulation -------------------------------%
%
TIME_DOMAIN= YES
TIME_MARCHING= DUAL_TIME_STEPPING-1ST_ORDER
TIME_STEP= 1e-3
INNER_ITER= 10
TIME_ITER= 2
%
% ----------- COMPRESSIBLE AND INCOMPRESSIBLE FREE-STREAM DEFINITION ----------%
%
FLUID_MODEL= IDEAL_GAS 
GAMMA_VALUE= 1.4
GAS_CONSTANT= 287.0
MACH_NUMBER= 0.5
AOA= 0.0
SIDESLIP_ANGLE= 0.0
FREESTREAM_TEMPERATURE= 236.2152
FREESTREAM_PRESSURE=3.565153e+004
%REYNOLDS_NUMBER= 200.0
%REYNOLDS_LENGTH= 1.0

% ----------------------- DYNAMIC MESH DEFINITION -----------------------------%
%
SURFACE_MOVEMENT= (EXTERNAL, EXTERNAL)
MARKER_MOVING= ( wall_up, wall_dw )
% ------------------------ GRID DEFORMATION PARAMETERS ------------------------%
%
DEFORM_LINEAR_SOLVER= CONJUGATE_GRADIENT
DEFORM_LINEAR_SOLVER_PREC= LU_SGS
DEFORM_LINEAR_SOLVER_ITER= 1000
DEFORM_NONLINEAR_ITER= 1
DEFORM_CONSOLE_OUTPUT= YES
DEFORM_LINEAR_SOLVER_ERROR= 1E-14
DEFORM_COEFF = 1E3
DEFORM_STIFFNESS_TYPE= WALL_DISTANCE
% ---------------------- REFERENCE VALUE DEFINITION ---------------------------%
%
REF_ORIGIN_MOMENT_X = 0.00
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
REF_LENGTH= 1.0
REF_AREA= 1.0

% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
%
MARKER_EULER= ( wall_up, wall_dw )
MARKER_SUPERSONIC_OUTLET= ( OUTLET )
INLET_TYPE= MASS_FLOW
MARKER_INLET = (INLET, 0.57 , 154.038, 0.0, 1.0, 0.0)
MARKER_PLOTTING= ( wall_up, wall_dw )
MARKER_MONITORING= ( wall_up, wall_dw )
MARKER_SYM = ( SYMMETRY )

% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------%
%
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES
CFL_NUMBER= 5.0
CFL_ADAPT= NO
CFL_ADAPT_PARAM= ( 1.5, 0.5, 1.0, 100.0 )
%ITER= 99999

% ----------------------- SLOPE LIMITER DEFINITION ----------------------------%
%
VENKAT_LIMITER_COEFF= 0.03
ADJ_SHARP_LIMITER_COEFF= 3.0
REF_SHARP_EDGES= 3.0
SENS_REMOVE_SHARP= NO

% ------------------------ LINEAR SOLVER DEFINITION ---------------------------%
%
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= LU_SGS
LINEAR_SOLVER_ERROR= 1E-4
LINEAR_SOLVER_ITER= 5

% -------------------------- MULTIGRID PARAMETERS -----------------------------%
%
MGLEVEL= 3
MGCYCLE= V_CYCLE
MG_PRE_SMOOTH= ( 1, 1, 1, 1 )
MG_POST_SMOOTH= ( 0, 0, 0, 0 )
MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
MG_DAMP_RESTRICTION= 0.5
MG_DAMP_PROLONGATION= 0.5

% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
%
CONV_NUM_METHOD_FLOW= HLLC
MUSCL_FLOW= NO
SLOPE_LIMITER_FLOW= NONE
JST_SENSOR_COEFF= ( 0.5, 0.02 )
TIME_DISCRE_FLOW= EULER_IMPLICIT

% --------------------------- CONVERGENCE PARAMETERS --------------------------%
%
CONV_RESIDUAL_MINVAL= -15
CONV_STARTITER= 10
CONV_CAUCHY_ELEMS= 100
CONV_CAUCHY_EPS= 1E-7

% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
%

MESH_FILENAME= fluid.su2
MESH_FORMAT= SU2
MESH_OUT_FILENAME= mesh_out.su2
SOLUTION_FILENAME= restart_flow.csv
SOLUTION_ADJ_FILENAME= solution_adj.dat
TABULAR_FORMAT= CSV
CONV_FILENAME= history
RESTART_FILENAME= restart_flow.dat
RESTART_ADJ_FILENAME= restart_adj.dat
VOLUME_FILENAME= flow
VOLUME_ADJ_FILENAME= adjoint
GRAD_OBJFUNC_FILENAME= of_grad.dat
SURFACE_FILENAME= surface_flow
SURFACE_ADJ_FILENAME= surface_adjoint
SCREEN_OUTPUT= (INNER_ITER, RMS_DENSITY, RMS_ENERGY, LIFT, DRAG)
OUTPUT_FILES= ( RESTART_ASCII, PARAVIEW, SURFACE_PARAVIEW)
OUTPUT_WRT_FREQ=(100,100,100)

**Comment by pcarruscag** [comment]
Can you provide all the required files to reproduce the issue please?

**Comment by cucss77** [comment]
> Can you provide all the required files to reproduce the issue please?

I removed the restart. I used mpiexec -n 9 SU2_CFD 

[SU2_ALE.zip](https://github.com/user-attachments/files/18267869/SU2_ALE.zip)

**Comment by pcarruscag** [comment]
You're lucky it's the time of giving...

I commented out the grid velocities to get the solver to output something, this is what it looks like:
![image](https://github.com/user-attachments/assets/1c12b9dc-5611-4f93-a7a1-4b459d8261a7)

There are 7666 unique node IDS used by the markers you are deforming, and your deformation file has 7663 lines, so you are missing data. Unfortunately whoever wrote this external deformation stuff didn't think of building in any sanity checks for the input data.

Your general setup is also very strange, initial flow direction doesn't match the boundary conditions, supersonic outlet with subsonic flow, etc. Here is a revised version.
[inv_wall.txt](https://github.com/user-attachments/files/18268356/inv_wall.txt)

**Reply by cucss77** [reply]
Yes, I know that this setting is strange. I'm developing a new technique for the unsteady FSI in which the CFD part is handled by SU2. 

I modified the .cfg file in according with your text and I fixed the node number differences. I get this error:

# CG residual history
# Residual tolerance target = 1e-14
# Initial residual norm     = nan
     0     nan
     10     nan
     20     nan
     30     nan
     40     nan
     50     nan
     60     nan
     70     nan
     80     nan
     90     nan
     .
     .
     .
     
     I think that the problem could be in the computational mesh. Are you agree?
